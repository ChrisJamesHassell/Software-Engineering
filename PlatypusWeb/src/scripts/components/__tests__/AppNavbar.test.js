import React from 'react';
import { shallow, mount, render } from 'enzyme';
import AppNavbar from '../Navbar/AppNavbar';


describe('AppNavbar', () => {

  it('matches the snapshot', () =>{
      const tree = shallow(<AppNavbar />)
      expect(tree).toMatchSnapshot()
  })
})