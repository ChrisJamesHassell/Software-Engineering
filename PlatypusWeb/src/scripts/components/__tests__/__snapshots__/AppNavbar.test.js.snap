// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavbar matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppNavbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": Array [
        <AppLogoHeader
          logo="logo_fill_white.svg"
        />,
        <NavbarCollapse>
          <Nav
            bsClass="nav"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
          >
            <NavDropdown
              eventKey={10}
              id="basic-nav-dropdown"
              title={
                <span
                  id="nav-profile"
                >
                  <span
                    id="nav-profile-username"
                  />
                </span>
              }
            >
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={10.1}
                header={false}
              >
                Something1
              </MenuItem>
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={10.2}
                header={false}
              >
                Something2
              </MenuItem>
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={true}
                header={false}
              />
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={10.3}
                header={false}
              >
                Something3
              </MenuItem>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                style={
                  Object {
                    "borderRadius": "0",
                    "width": "100%",
                  }
                }
              >
                Sign out
              </Button>
            </NavDropdown>
          </Nav>
        </NavbarCollapse>,
      ],
      "collapseOnSelect": true,
      "style": Object {
        "borderRadius": "0",
        "display": "none",
        "marginBottom": "0",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "logo": "logo_fill_white.svg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Nav
            bsClass="nav"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
          >
            <NavDropdown
              eventKey={10}
              id="basic-nav-dropdown"
              title={
                <span
                  id="nav-profile"
                >
                  <span
                    id="nav-profile-username"
                  />
                </span>
              }
            >
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={10.1}
                header={false}
              >
                Something1
              </MenuItem>
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={10.2}
                header={false}
              >
                Something2
              </MenuItem>
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={true}
                header={false}
              />
              <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                eventKey={10.3}
                header={false}
              >
                Something3
              </MenuItem>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                style={
                  Object {
                    "borderRadius": "0",
                    "width": "100%",
                  }
                }
              >
                Sign out
              </Button>
            </NavDropdown>
          </Nav>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "nav",
            "children": Array [
              undefined,
              undefined,
              <NavDropdown
                eventKey={10}
                id="basic-nav-dropdown"
                title={
                  <span
                    id="nav-profile"
                  >
                    <span
                      id="nav-profile-username"
                    />
                  </span>
                }
              >
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.1}
                  header={false}
                >
                  Something1
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.2}
                  header={false}
                >
                  Something2
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={true}
                  header={false}
                />
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.3}
                  header={false}
                >
                  Something3
                </MenuItem>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "borderRadius": "0",
                      "width": "100%",
                    }
                  }
                >
                  Sign out
                </Button>
              </NavDropdown>,
            ],
            "justified": false,
            "pullLeft": false,
            "pullRight": true,
            "stacked": false,
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey={10.1}
                    header={false}
                  >
                    Something1
                  </MenuItem>,
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey={10.2}
                    header={false}
                  >
                    Something2
                  </MenuItem>,
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={true}
                    header={false}
                  />,
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey={10.3}
                    header={false}
                  >
                    Something3
                  </MenuItem>,
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "borderRadius": "0",
                        "width": "100%",
                      }
                    }
                  >
                    Sign out
                  </Button>,
                ],
                "eventKey": 10,
                "id": "basic-nav-dropdown",
                "title": <span
                  id="nav-profile"
                >
                  <span
                    id="nav-profile-username"
                  />
                </span>,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "dropdown",
                    "children": "Something1",
                    "disabled": false,
                    "divider": false,
                    "eventKey": 10.1,
                    "header": false,
                  },
                  "ref": null,
                  "rendered": "Something1",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "dropdown",
                    "children": "Something2",
                    "disabled": false,
                    "divider": false,
                    "eventKey": 10.2,
                    "header": false,
                  },
                  "ref": null,
                  "rendered": "Something2",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "dropdown",
                    "disabled": false,
                    "divider": true,
                    "header": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "dropdown",
                    "children": "Something3",
                    "disabled": false,
                    "divider": false,
                    "eventKey": 10.3,
                    "header": false,
                  },
                  "ref": null,
                  "rendered": "Something3",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "default",
                    "children": "Sign out",
                    "disabled": false,
                    "onClick": [Function],
                    "style": Object {
                      "borderRadius": "0",
                      "width": "100%",
                    },
                  },
                  "ref": null,
                  "rendered": "Sign out",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <AppLogoHeader
            logo="logo_fill_white.svg"
          />,
          <NavbarCollapse>
            <Nav
              bsClass="nav"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
            >
              <NavDropdown
                eventKey={10}
                id="basic-nav-dropdown"
                title={
                  <span
                    id="nav-profile"
                  >
                    <span
                      id="nav-profile-username"
                    />
                  </span>
                }
              >
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.1}
                  header={false}
                >
                  Something1
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.2}
                  header={false}
                >
                  Something2
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={true}
                  header={false}
                />
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.3}
                  header={false}
                >
                  Something3
                </MenuItem>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "borderRadius": "0",
                      "width": "100%",
                    }
                  }
                >
                  Sign out
                </Button>
              </NavDropdown>
            </Nav>
          </NavbarCollapse>,
        ],
        "collapseOnSelect": true,
        "style": Object {
          "borderRadius": "0",
          "display": "none",
          "marginBottom": "0",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "logo": "logo_fill_white.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Nav
              bsClass="nav"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
            >
              <NavDropdown
                eventKey={10}
                id="basic-nav-dropdown"
                title={
                  <span
                    id="nav-profile"
                  >
                    <span
                      id="nav-profile-username"
                    />
                  </span>
                }
              >
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.1}
                  header={false}
                >
                  Something1
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.2}
                  header={false}
                >
                  Something2
                </MenuItem>
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={true}
                  header={false}
                />
                <MenuItem
                  bsClass="dropdown"
                  disabled={false}
                  divider={false}
                  eventKey={10.3}
                  header={false}
                >
                  Something3
                </MenuItem>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "borderRadius": "0",
                      "width": "100%",
                    }
                  }
                >
                  Sign out
                </Button>
              </NavDropdown>
            </Nav>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "nav",
              "children": Array [
                undefined,
                undefined,
                <NavDropdown
                  eventKey={10}
                  id="basic-nav-dropdown"
                  title={
                    <span
                      id="nav-profile"
                    >
                      <span
                        id="nav-profile-username"
                      />
                    </span>
                  }
                >
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey={10.1}
                    header={false}
                  >
                    Something1
                  </MenuItem>
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey={10.2}
                    header={false}
                  >
                    Something2
                  </MenuItem>
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={true}
                    header={false}
                  />
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey={10.3}
                    header={false}
                  >
                    Something3
                  </MenuItem>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "borderRadius": "0",
                        "width": "100%",
                      }
                    }
                  >
                    Sign out
                  </Button>
                </NavDropdown>,
              ],
              "justified": false,
              "pullLeft": false,
              "pullRight": true,
              "stacked": false,
            },
            "ref": null,
            "rendered": Array [
              undefined,
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <MenuItem
                      bsClass="dropdown"
                      disabled={false}
                      divider={false}
                      eventKey={10.1}
                      header={false}
                    >
                      Something1
                    </MenuItem>,
                    <MenuItem
                      bsClass="dropdown"
                      disabled={false}
                      divider={false}
                      eventKey={10.2}
                      header={false}
                    >
                      Something2
                    </MenuItem>,
                    <MenuItem
                      bsClass="dropdown"
                      disabled={false}
                      divider={true}
                      header={false}
                    />,
                    <MenuItem
                      bsClass="dropdown"
                      disabled={false}
                      divider={false}
                      eventKey={10.3}
                      header={false}
                    >
                      Something3
                    </MenuItem>,
                    <Button
                      active={false}
                      block={false}
                      bsClass="btn"
                      bsStyle="default"
                      disabled={false}
                      onClick={[Function]}
                      style={
                        Object {
                          "borderRadius": "0",
                          "width": "100%",
                        }
                      }
                    >
                      Sign out
                    </Button>,
                  ],
                  "eventKey": 10,
                  "id": "basic-nav-dropdown",
                  "title": <span
                    id="nav-profile"
                  >
                    <span
                      id="nav-profile-username"
                    />
                  </span>,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "dropdown",
                      "children": "Something1",
                      "disabled": false,
                      "divider": false,
                      "eventKey": 10.1,
                      "header": false,
                    },
                    "ref": null,
                    "rendered": "Something1",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "dropdown",
                      "children": "Something2",
                      "disabled": false,
                      "divider": false,
                      "eventKey": 10.2,
                      "header": false,
                    },
                    "ref": null,
                    "rendered": "Something2",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "dropdown",
                      "disabled": false,
                      "divider": true,
                      "header": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "dropdown",
                      "children": "Something3",
                      "disabled": false,
                      "divider": false,
                      "eventKey": 10.3,
                      "header": false,
                    },
                    "ref": null,
                    "rendered": "Something3",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "block": false,
                      "bsClass": "btn",
                      "bsStyle": "default",
                      "children": "Sign out",
                      "disabled": false,
                      "onClick": [Function],
                      "style": Object {
                        "borderRadius": "0",
                        "width": "100%",
                      },
                    },
                    "ref": null,
                    "rendered": "Sign out",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
